plugins/arCasPlugin/lib/arCAS.class.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * This file is based heavily on the sfCASPlugin developed by D.Jeanmonod and
+ * maintained by H.Lepesant, MIT License, https://github.com/jeanmonod/sfCASPlugin.
+ */
+
+class arCAS
+{
+  protected static $phpCASIsInitialized = false;
+
+  /**
+   * Initialize the phpCAS library
+   */
+  public static function initializePhpCAS()
+  {  
+    // Return if phpCAS is already initialized
+    if (self::$phpCASIsInitialized) {
+      return;
+    }
+
+    require_once sfConfig::get('sf_root_dir').'/vendor/composer/jasig/phpcas/CAS.php';
+
+    if (sfConfig::get('sf_debug', false) == true)
+    {
+      $debugLogPath = sfConfig::get('sf_log_dir').'/phpcas.log';
+      phpCAS::setDebug($debugLogPath);
+      phpCAS::setVerbose(true);
+    }
+
+    $casVersion = sfConfig::get('app_cas_cas_version', '3.0');
+    $validCasVersions = array('1.0', '2.0', '3.0', 'S1');
```

Comment:
My understanding of the CAS standard and the phpcas library is that there's good support for all four versions. I've tested with `1.0`, `2.0` and `3.0` during development and confirmed that login/logout works as expected across all versions. They're quite similar, with the main difference being that CAS user attributes are only available in `3.0`. It [looks like `S1`adds some additional methods we're not making use of](https://apereo.github.io/phpCAS/api/group__internalSAML.html), so it might be safe to drop that one.

Reply:
I removed `S1` as one of the valid CAS version options in https://github.com/artefactual/atom/pull/1176/commits/8c00e4c493746aa9fca7bbea458d402e3e87eb97.